subsys/net/lib/download_client/src/http_download_client.c

```diff
@@ -49,71 +56,126 @@ static int socket_timeout_set(int fd)
 	return 0;
 }
 
-static int resolve_and_connect(const char *const host, const char *const port,
-			       u32_t family, u32_t proto)
+static int socket_sectag_set(int fd, int sec_tag)
+{
+	int err;
+	int verify;
+	sec_tag_t sec_tag_list[] = { sec_tag };
+
+	enum {
+		NONE = 0,
+		OPTIONAL = 1,
+		REQUIRED = 2,
+	};
+
+	verify = OPTIONAL;
+
+	err = setsockopt(fd, SOL_TLS, TLS_PEER_VERIFY, &verify, sizeof(verify));
+	if (err) {
+		LOG_ERR("Failed to setup peer verification, errno %d", errno);
+		return -1;
+	}
+
+	err = setsockopt(fd, SOL_TLS, TLS_SEC_TAG_LIST, sec_tag_list,
+			 sizeof(sec_tag_t) * ARRAY_SIZE(sec_tag_list));
+	if (err) {
+		LOG_ERR("Failed to setup socket security tag, errno %d", errno);
+		return -1;
+	}
+	return 0;
+}
+
+static int resolve_and_connect(const char *const host, int family, int sec_tag)
 {
 	int fd;
-	int rc;
+	int err;
+	int proto;
+	u16_t port;
+	struct addrinfo *addr;
+	struct addrinfo *info;
 
 	if (host == NULL) {
 		return -1;
 	}
 
-	struct addrinfo *addrinf = NULL;
+	/* Set up port and protocol */
+	if (sec_tag < 0) {
+		/* HTTP, port 80 */
+		proto = IPPROTO_TCP;
+		port = htons(80);
+	} else {
+		/* HTTPS, port 443 */
+		proto = IPPROTO_TLS_1_2;
+		port = htons(443);
+	}
+
+	/* Lookup  host */
 	struct addrinfo hints = {
 		.ai_family = family,
 		.ai_socktype = SOCK_STREAM,
 		.ai_protocol = proto,
 	};
 
-	/* Resolve the port. */
-	rc = getaddrinfo(host, port, &hints, &addrinf);
-
-	if (rc < 0 || (addrinf == NULL)) {
-		LOG_ERR("getaddrinfo() failed, err %d", errno);
+	err = getaddrinfo(host, NULL, &hints, &info);
+	if (err) {
+		LOG_WRN("Failed to resolve hostname %s on %s", host,
+			family == AF_INET ? "IPv4" : "IPv6");
 		return -1;
 	}
 
-	struct addrinfo *addr = addrinf;
-	struct sockaddr *remoteaddr;
-
-	int addrlen = (family == AF_INET6) ? sizeof(struct sockaddr_in6) :
-					     sizeof(struct sockaddr_in);
+	LOG_INF("Attempting to connect over %s",
+		family == AF_INET ? "IPv4" : "IPv6");
 
-	/* Open a socket based on the local address. */
 	fd = socket(family, SOCK_STREAM, proto);
-	if (fd >= 0) {
-		/* Look for IPv4 address of the broker. */
-		while (addr != NULL) {
-			remoteaddr = addr->ai_addr;
-
-			if (remoteaddr->sa_family == family) {
-				((struct sockaddr_in *)remoteaddr)->sin_port =
-					htons(80);
-
-				/** TODO:
-				 *  Need to set security setting for HTTPS.
-				 */
-				rc = connect(fd, remoteaddr, addrlen);
-				if (rc == 0) {
-					break;
-				}
-			}
-			addr = addr->ai_next;
+	if (fd < 0) {
+		LOG_ERR("Failed to create socket, errno %d", errno);
+		goto cleanup;
+	}
+
+	if (proto == IPPROTO_TLS_1_2) {
+		LOG_DBG("Setting up TLS credentials");
+		err = socket_sectag_set(fd, sec_tag);
+		if (err) {
+			goto cleanup;
 		}
 	}
 
-	freeaddrinfo(addrinf);
+	/* Not connected */
+	err = -1;
 
-	if (rc < 0) {
+	for (addr = info; addr != NULL; addr = addr->ai_next) {
+
```

Comment:
No newline needed at the beginning of the block.

Reply:
I had added it for readability, removed now :)